tensorboard/util/tensor_manip.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities to create TensorProtos."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorboard.compat import tf
+
+# TODO(stephanwlee): move the implmentation of
+# tensorboard/compat/tensorflow_stubs/tensor_manip.pv to here.
```

Comment:
Do we want to just do this in this same PR?  Or did you want to wait on it?

When we move the implementation, we should probably copy over some of the tests from TF as well so that our now-independent implementation is still tested.

Reply:
Though sound easy, I found few quirks that it was a bit more challenging to have tf and notf both work. Specifically, there was no easy way for me to get this work:
https://github.com/tensorflow/tensorflow/blob/82e302a84e06342846ab809e1f0c0bdb9e0332a9/tensorflow/python/framework/tensor_shape.py#L1120-L1125